### ISSUE TITLE
Gauche: Cannot build stable on Cygwin

### ISSUE THREAD

**Issue by okuoku** [okuoku/yunibase#17]
```
gcc -DHAVE_CONFIG_H -I. -I. -I./../gc/include -I../gc/include -I./../gc/libatomic_ops/src -I../gc/libatomic_ops/src   -g -O2  -fomit-frame-pointer  -c libsys.c
libsys.scm: 関数 ‘libsyssys_ctermid’ 内:
libsys.scm:415:12: エラー: ‘L_ctermid’ が宣言されていません (この関数内での最初の使用)
        (let* ([buf::(.array char [(+ L_ctermid 1)])])
            ^
libsys.scm:415:12: 備考: 未宣言の識別子は出現した各関数内で一回のみ報告されます
```

It is highly likely something wrong on recent Cygwin update.

On Cygwin `<stdio.h>`:

``` c
#if __XSI_VISIBLE
#define L_cuserid       9               /* posix says it goes in stdio.h :( */
#ifdef __CYGWIN__
#define L_ctermid       16 ★
#endif
#endif
```

Feature macro `__XSI_VISIBLE` seems just be wrong; we could use POSIX_SOURCE etc here.

**Comment by okuoku** [comment]
The change was:

https://cygwin.com/git/gitweb.cgi?p=newlib-cygwin.git;a=commit;h=feec81e5717f904b687f17ace727bfbadc597209

**Comment by okuoku** [comment]
Upstream fixed the problem. I'd forget file a bug for @shirok :P
https://cygwin.com/ml/cygwin/2016-05/msg00279.html

**Comment by shirok** [comment]
So, does this upstream fix address the issue?  Or does Gauche needs to take an action?

**Comment by okuoku** [comment]
> So, does this upstream fix address the issue?

I think upcoming Cygwin release should address the issue. We could `#ifdef L_ctermid` for fix it locally, but it turned out it is not necessary now (unless we cannot wait for a new Cygwin release -- for a few days or weeks). I wasn't sure upstream has any interest to fix this...

**Comment by okuoku** [comment]
Not only [Cygwin is now LGPL3 and commercial licensing dropped](https://cygwin.com/ml/cygwin/2016-06/msg00326.html), it also fixed the issue with [2.5.2, the last release that supports WinXP](https://cygwin.com/ml/cygwin/2016-06/msg00328.html). Drama.

Anyway, nothing to do more. CLOSE.
